File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_left(int space, int length, int row);

// main function
int main(void)
{
    int n;
    do
    {
        n = get_int("row: ");
    }
    while (n < 1 || n > 8);

    for (int i = 0; i < n; i++)
    {
        print_left(n - i - 1, i + 1, i + 1);
    }
}

void print_left(int space, int length, int row)
{
    for (int d = 0; d < space; d++)
    {
        printf(" ");
    }
    for (int i = 0; i < length; i++)
    {
        printf("#");
    }
    for (int e = 0; e < 2; e++)
    {
        printf(" ");
    }
    for (int i = 0; i < row; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: CS50X/sentimental-mario-more/mario.py
# we take the get_int function from the cs50 libary
from cs50 import get_int

while True:  # we make a loop
    n = get_int("Hight: ")
    if (n > 1 and n < 8):
        break
    else:
        pass


def main():
    for i in range(n):
        length(n - i - 1, i + 1, i + 1)

# length is function that took argument as space and length


def length(space, left_pyramid, right_pyramid):
    for j in range(space):  # it is for for every space
        print(" ", end="")

    for i in range(left_pyramid):  # hash
        print("#", end="")

    for d in range(2):  # it is for for every space
        print(" ", end="")

    for f in range(right_pyramid):
        print("#", end="")
    print()


main()
